<launch>
<group ns="map_merge">
  <node pkg="map_merge_3d" type="map_merge_node" respawn="false" name="map_merge_3d" output="screen">
    <param name="robot_map_topic" value="map"/>
    <param name="robot_namespace" value=""/>
    <param name="merged_map_topic" value="map"/>
    <param name="world_frame" value="world"/>
    <param name="compositing_rate" value="0.3"/>
    <param name="discovery_rate" value="0.05"/>
    <param name="estimation_rate" value="0.01"/>
    <param name="publish_tf" value="true"/>
    <!-- and all the other map merging parameters -->
    <param name="resolution" value="0.05"/>
    <param name="descriptor_type" value="SHOT" /> <!--PFH, PFHRGB, FPFH, RSD, SHOT, SC3D -->
    <param name="matching_k" value="5" />
    <param name="confidence_threshold" value="0.0"/>
    <param name="output_resolution" value="0.05"/>
    <param name="estimation_method" value="MATCHING"/> <!-- MATCHING, SAC_IA, NDT-->
    <param name="correspondence_method" value="KDTREE"/> <!-- KDTREE, RECIPROCAL-->
    <param name="keypoint_type" value="SIFT" /> <!-- HARRIS, SIFT, ISS, NARF-->
  </node>
</group>

<!-- 2D & 3D octomap mapper node-->
<node pkg="map_merge_3d" type="octomap_mapper_node" respawn="true" name="octomap_mapper" output="screen">
  <param name="resolution" value="0.1"/>
  <param name="merged_map_topic" value="map_merge/map"/>
</node>

</launch>
